src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClients/FailOverSupportedConfigurationClient.cs

```diff
@@ -0,0 +1,256 @@
+            }
+
+            this._clients = endpoints.Select(endpoint => new ConfigurationClientState(endpoint, new ConfigurationClient(endpoint, credential, options.ClientOptions))).ToList();
+            this._parallelRetryInterval = options.ParallelRetryInterval;
+        }
+
+        public Task<Response<ConfigurationSetting>> GetConfigurationSettingAsync(string key, string label = null, CancellationToken cancellationToken = default)
+        {
+            Task<Response<ConfigurationSetting>> func(ConfigurationClient client)
+            {
+                return client.GetConfigurationSettingAsync(key, label, cancellationToken);
+            }
+
+            return ExecuteWithFailOverPolicyAsync(func, cancellationToken);
+        }
+
+        public Task<Response<ConfigurationSetting>> GetConfigurationSettingAsync(ConfigurationSetting setting, bool onlyIfChanged = false, CancellationToken cancellationToken = default)
+        {
+            Task<Response<ConfigurationSetting>> func(ConfigurationClient client)
+            {
+                return client.GetConfigurationSettingAsync(setting, onlyIfChanged, cancellationToken);
+            }
+
+            return ExecuteWithFailOverPolicyAsync(func, cancellationToken);
+        }
+
+        public async Task<IEnumerable<ConfigurationSetting>> GetConfigurationSettingsAsync(SettingSelector selector, CancellationToken cancellationToken = default)
+        {
+            Page<ConfigurationSetting> settingsPage = null;
+            string continuationToken = null;
+            var result = new List<ConfigurationSetting>();
+
+            async Task<Page<ConfigurationSetting>> func(ConfigurationClient client)
+            {
+                IAsyncEnumerator<Page<ConfigurationSetting>> enumerator = client.GetConfigurationSettingsAsync(selector, cancellationToken)
+                                                                                .AsPages(continuationToken: continuationToken)
+                                                                                .GetAsyncEnumerator();
+
+                Page<ConfigurationSetting> page = enumerator != null && await enumerator.MoveNextAsync() ? enumerator.Current : null;
+                return page;
+            }
+
+            do
+            {
+                settingsPage = await ExecuteWithFailOverPolicyAsync(func, cancellationToken);
+                if (settingsPage != null)
+                {
+                    result.AddRange(settingsPage.Values);
+                    continuationToken = settingsPage.ContinuationToken;
+                }
+            }
+            while (settingsPage != null && settingsPage.ContinuationToken != null);
+
+            return result;
+        }
+
+        public void UpdateSyncToken(Uri endpoint, string syncToken)
+        {
+            this._clients.Single(clientAndState => clientAndState.Endpoint.Host.Equals(endpoint.Host)).Client.UpdateSyncToken(syncToken);
```

Comment:
Can the sync token be shared across replicas?

I am wondering if it is possible the endpoint included in the push notification may not be included in the list of endpoints a user passed in.

Reply:
I'm not sure if the sync token can be shared across replicas. @jimmyca15 do you know?

Reply:
It doesn't make sense to share sync tokens across replicas. It won't hurt, other than taking up memory, but it's unnecessary.

> I am wondering if it is possible the endpoint included in the push notification may not be included in the list of endpoints a user passed in.

Our events do contain the request URI that was used to update key-value.

Reply:
Here is my scenario:

A user makes a key-value change in region 1. The application has only endpoints of region 2 and 3. How the application can make sure it gets the updated key-value after receiving the push notification?

Reply:
That's a good point, but I'd simplify it even further. Say the application has endpoints of region 1 and 2, and they update a key-value in region 2. It seems the best we could do with what we have is on next request, attempt to use endpoint 2 first. However, endpoint 2 could fail and we would serve with endpoint 1 which may not have a change. It needs some thought.